README.md

```diff
@@ -1,89 +1,119 @@
+This repository represents the 5GMS Application Provider side in a 3GPP compliant 5G Media Streaming
+architecture. It interacts with the [5GMS Application Function](https://github.com/5G-MAG/rt-5gms-application-function)
+via the interface at reference point M1. For that reason, this repository comes with multiple different tools:
+
+* **CLI tool**: The Python-based Command Line Interface tool for 5GMS management is a set of executable wrapper modules built upon Python classes which interacts with the 5GMS Application Function's RESTful API at reference point M1 to provision 5GMS services. 
+* **Management UI**: This is a full stack application which utilizes the Python executable modules to interact with the 5GMS Application Function. Every M1 provisioning procedure is implemented as a corresponding web-server endpoint, supported with the graphical control dashboard.
+* **Postman**: Postman recipes to test the 5GMS Application Function's  API at reference point M1. This is a collection of predefined HTTP requests for every Application Function's RESTful endpoint, including environmental variables and requests payload.
+
+## Installing the Command Line Interface tool
+
+In order to use Python CLI tool to control 5GMS Application Function operations, compile `python` library:
+
+```
+cd rt-5gms-application-provider
+python3 -m pip install ./python
+```
+
+This will install the executables i.e. `m1-session`, `m1-client` and `msaf-configuration` and associated `rt_m1_client`
+python module.
+For testing use a venv so that you don't change system python modules, e.g.:
+
+```
+cd rt-5gms-application-provider
+python3 -m venv venv
+venv/bin/python3 -m pip install ./python
+```
+
+## Installing the Management UI
+
+There are two ways to run this application. The recommendation is to use the **Docker Compose** service, because
+lightweight container building and activation will solve the entire scope of dependencies and the rest of the
+requirements.
+
+### 1. Docker Compose
 
-# Building and installing
-There are two ways to run this project, and the warm reccomendation is to use **Docker Compose** service, because lightweight container building and activation will solve entire scope of dependecies and the rest of requirements.
+To install this service follow the official [documentation](https://docs.docker.com/compose/install/). Next, clone this
+repository:
 
-## Docker Compose
-To install this service follow the official [documentation](https://docs.docker.com/compose/install/). Next, clone this repository:
 ```
 cd
 git clone https://github.com/5G-MAG/rt-5gms-application-provider
 cd ~/rt-5gms-application-provider
 ```
 
-Building Docker image will effectivelly install all dependecies for RT Application Function and the rest of Python dependencies:
+Building the Docker image will effectively install all dependencies for
+the 5GMS Application Function and this application:
```

Comment:
what is "this application"? if it refers to the Application Provider I would state it to avoid confusion